platforms/documentation/docs/src/docs/userguide/best-practices/bp_general.adoc

```diff
@@ -15,4 +15,36 @@
 [[best_practices_general]]
 = General Gradle Best Practices
 
-Add content here.
+== Do Not Use Internal APIs
+
+Do not use APIs that are inside an `internal` Java package or contain `Internal` or `Impl`
+in the name of the type.
+
+=== Explanation
+
+Using internal APIs is inherently risky and can cause significant problems during upgrades.
```

Comment:
💅 Formatting note: Please start each sentence on a new line in a `.md` or `.adoc` file.  Separate paragraphs with a blank line.